src/Tests/Microsoft.NET.TestFramework/Attributes/RequiresMSBuildVersionFactAttribute.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Xunit;
+
+namespace Microsoft.NET.TestFramework
+{
+    public class RequiresMSBuildVersionFactAttribute : FactAttribute
+    {
+        public RequiresMSBuildVersionFactAttribute(string version)
+        {
+            if (!Version.TryParse(TestContext.Current.ToolsetUnderTest.MSBuildVersion, out Version msbuildVersion))
```

Comment:
I think these attributes should never skip if running on .NET Core (if we don't have the required MSBuild version in the .NET SDK, then we wouldn't be writing the test at all).  If running on full framework, then we need to shell out to `msbuild` instead of `dotnet msbuild` to get the full framework MSBuild version.  Also, make sure that we handle 3 versus 4 part version numbers correctly (where the required version is 3 parts and the msbuild version is 4 parts).